go/chat/utils/collapses.go

```diff
@@ -0,0 +1,117 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/clockwork"
+	"golang.org/x/net/context"
+)
+
+type singleCollapseRecord struct {
+	Collapsed bool
+	Time      time.Time
+}
+
+type rangeCollapseRecord struct {
+	Collapsed bool
+	MsgID     chat1.MessageID
+	Time      time.Time
+}
+
+type Collapses struct {
+	globals.Contextified
+	DebugLabeler
+
+	clock clockwork.Clock
+}
+
+func NewCollapses(g *globals.Context) *Collapses {
```

Comment:
should htis have an in mem cache as well?

Reply:
Didn't want to bother with it, it shouldn't be called too often.